frontend/src/common/queries/wheresMyGene.ts

```diff
@@ -186,6 +190,7 @@ async function fetchQuery(
     ...JSON_BODY_FETCH_OPTIONS,
     body: JSON.stringify(query),
     method: "POST",
+    signal,
```

Comment:
this tells fetch API to abort the request on the fly when the react-query is canceled!

https://react-query.tanstack.com/guides/query-cancellation